app/src/user/user.service.ts

```diff
@@ -336,10 +358,58 @@ export class UserService {
             }
         }`;
 
+        const removeUserTechnologies = `mutation {
+            delete_user_technology(
+                where: {
+                    user_id: {_eq: "${userId}"}
+                },
+            ) {
+                returning {
+                    user_id
+                    technology_id
+                }
+            }
+        }`;
+
+        let userTechnologiesTemp = [];
+        if (technologies.length) {
+            userTechnologiesTemp = technologies.map(el => {
+                return `{
+                    user_id: "${userId}",
+                    technology_id: "${el}",
+                }`;
+            });
+        }
+
+        const insertUserTechnologies = `mutation {
+            insert_user_technology(
+                objects: [${userTechnologiesTemp}]
+            ) {
+                returning {
+                    user_id
+                    technology_id
+                }
+            }
+        }`;
+
         return new Promise(async (resolve, reject) => {
             this.httpRequestsService
                 .request(query)
                 .subscribe((res: AxiosResponse) => resolve(res), (error: AxiosError) => reject(error));
+            this.httpRequestsService
```

Comment:
Lines from 399  to 412 is an uncontrolled chain of code. Place it into the single chain